engines/sci/engine/kvideo.cpp

```diff
@@ -136,10 +136,18 @@ reg_t kShowMovie(EngineState *s, int argc, reg_t *argv) {
 			// Mac QuickTime
 			// The only argument is the string for the video
 
-			// HACK: Switch to 16bpp graphics for Cinepak.
+			// Switch to 16bpp graphics for Cinepak
 			if (g_system->getScreenFormat().bytesPerPixel == 1) {
-				initGraphics(screenWidth, screenHeight, nullptr);
-				switchedGraphicsMode = true;
+				const Common::List<Graphics::PixelFormat> supportedFormats = g_system->getSupportedFormats();
+				Common::List<Graphics::PixelFormat>::const_iterator it;
+				for (it = supportedFormats.begin(); it != supportedFormats.end(); ++it) {
+					if (it->bytesPerPixel == 2) {
```

Comment:
Is there a reason why a 2bpp pixel format specifically is required here?

Reply:
The comment says "Switch to 16bpp graphics for Cinepak" so yes, it would appear that a reason existed. clone272 wrote this 12 years ago: cd5e90f298098fa44299bcf054fffeada9da2996

All I know about this code is that now the pixel format that's always been selected continues to be when upscaling, instead of crashing.

Reply:
This was added for the intro movie of KQ6 Mac, and the change done here is correct

Reply:
The reason I ask is that it would be preferable to allow using a 4bpp surface on platforms that prefer it, and was wondering if there was anything in particular in that stopped it from working.

(As an aside, if KQ6 Mac uses 24 bit Cinepak, RPZA or QTRLE, it might be neat to enable dithering when RGB mode is disabled, but that's something for another day.)